extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQSelectTest.java

```diff
@@ -2119,6 +2121,207 @@ public void testJoinUsesDifferentAlgorithm()
         .verifyResults();
   }
 
+  @Test
+  public void testSelectUnnestOnInlineFoo()
```

Comment:
Please add tests cases for insert and replace q's as well. 
They would be present in 
MSQInsertTests, MSQReplaceTests. 

I wonder if we should have some tests in which take in a select query and the framework makesthe insert and the replace query from it.

Reply:
The tests have been added